README.md

```diff
@@ -1,37 +1,74 @@
- * use or upgrade to the latest format version
+## [Model Description File Specification](./generated/bioimageio_model_spec.md)
 
+The [`Model Resource Description File Specification`](./generated/bioimageio_model_spec.md)(`model RDF`) defines a file format for representing pretrained AI models in [YAML format](https://en.wikipedia.org/wiki/YAML). This format is used to describe models hosted on the [BioImage.IO](https://bioimage.io) model repository site.
 
-# Model Specification Validator
-## Installation
+### Model Description File Examples
+ - [UNet 2D Nuclei Broad](https://github.com/bioimage-io/pytorch-bioimage-io/blob/master/specs/models/unet2d_nuclei_broad/UNet2DNucleiBroad.model.yaml).
+
+
+## [Generic Resource Description File Specification](./bioimageio_rdf_spec.md)
+
+A BioImage.IO-compatible Resource Description File (RDF) is a YAML file with a set of specifically defined fields. 
+
+The common fields for all the resource description files are:
+
+| Field | Required | Definition  |
+|---|---|---|
+| id | required | unique identifier for the resource, you can also use a `doi` e.g. generated from [zenodo](https://zenodo.org/) |
+| type | required | type of the resource, e.g. `model`, `dataset`, `application` or `notebook` |
+| name  | required | name of the resource, a human-friendly name  |
+| description | required | short description for the resource |
+| source | recommended | url to the source of the resource |
+| download_url | recommended | url to the zipped file if applicable |
+| git_repo | recommended | url to the git repo  |
+| license | recommended | the license name for the resource, e.g.: `MIT`  |
+| tags | recommended | a list of tags  |
+| covers | recommended | a list of url to the cover images (aspect ratio width/height=2/1)  |
+| attachments | optional | a group of attachments, list of resources grouped by keys  |
+| authors | optional | the full list of author names  |
+| badges | optional | a list of badges|
+| cite | optional | how to cite the resource |
+| documentation | optional | url to the documentation in markdown format  |
+| icon | optional | an icon for the resource  |
+| links | optional | a list of linked resources, an id to other resources|
+| version | optional | the version number for the resource, starting from `0.1.0`  |
+
+
+## Recommendations
+
+ * Describing AI models in the model-specific spec (i.e. [model RDF](./generated/bioimageio_model_spec.md)) instead of the generic RDF. Only fallback to the generic RDF if writing model specific RDF is not possible for some reason.
+ * The RDF or package file name should not space or special characters, it should be concise, descriptive, in kebab case or camel case.
+ * Due to the limitations of storage services such as Zenodo which does not support subfolders, it is recommended to place other files in the same directory level of the RDF file and try to avoid using subdirectories.
```

Comment:
```suggestion
 * Due to the limitations of storage services such as Zenodo, which does not support subfolders, it is recommended to place other files in the same directory level of the RDF file and try to avoid using subdirectories.
```